src/hidpi.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright © 2015, Alex Daniel Jones
+// Licensed under the MIT License <LICENSE.md>
+// Taken from hidpi.h
+
+#[repr(C)] #[derive(Clone, Copy, Debug)]
+pub enum HIDP_REPORT_TYPE {
+  HidP_Input,
+  HidP_Output,
+  HidP_Feature,
+}
+
+#[repr(C)] #[derive(Clone, Copy, Debug)]
+pub struct USAGE_AND_PAGE {
+    pub Usage: ::USAGE,
+    pub UsagePage: ::USAGE,
+}
+pub type PUSAGE_AND_PAGE = *mut USAGE_AND_PAGE;
+
+#[repr(C)] #[derive(Clone, Copy, Debug)]
+pub struct HIDP_BUTTON_CAPS {
+    pub UsagePage: ::USAGE,
+    pub ReportID: ::UCHAR,
+    pub IsAlias: ::BOOLEAN,
+
+    pub BitField: ::USHORT,
+    pub LinkCollection: ::USHORT,
+
+    pub LinkUsage: ::USAGE,
+    pub LinkUsagePage: ::USAGE,
+
+    pub IsRange: ::BOOLEAN,
+    pub IsStringRange: ::BOOLEAN,
+    pub IsDesignatorRange: ::BOOLEAN,
+    pub IsAbsolute: ::BOOLEAN,
+
+    pub Reserved: [::ULONG; 10],
+    pub Range: RANGE_STRUCT,
+}
+pub type PHIDP_BUTTON_CAPS = *mut HIDP_BUTTON_CAPS;
+
+#[repr(C)] #[derive(Clone, Copy, Debug)]
+pub struct RANGE_STRUCT {
+    pub UsageMin: ::USAGE,
+    pub UsageMax: ::USAGE,
+    pub StringMin: ::USHORT,
+    pub StringMax: ::USHORT,
+    pub DesignatorMin: ::USHORT,
+    pub DesignatorMax: ::USHORT,
+    pub DataIndexMin: ::USHORT,
+    pub DataIndexMax: ::USHORT,
+}
+
+#[repr(C)] #[derive(Clone, Copy, Debug)]
+pub struct HIDP_VALUE_CAPS {
+    pub UsagePage: ::USAGE,
+    pub ReportID: ::UCHAR,
+    pub IsAlias: ::BOOLEAN,
+
+    pub BitField: ::USHORT,
```

Comment:
In this particular case it looks like it's just a variable name: https://msdn.microsoft.com/en-us/library/windows/hardware/ff539832(v=vs.85).aspx however HIDP_LINK_COLLECTION_NODE  has bitfields and has to be fixed. Also I should add some UNION! macros here (very handy). Wished I knew about this macro when I was using RAWINPUT.